R/recipes.R

```diff
@@ -104,25 +106,22 @@ model_lin_recipe <- function(data, pred_vars, cat_vars,
       starts_with("ccao_is"),
       fn = as.numeric
     ) %>%
-    step_impute_median(all_numeric(), -all_outcomes(), -has_role("ID")) %>%
+    step_impute_median(all_numeric_predictors(), -has_role("ID")) %>%
+    step_impute_mode(all_nominal_predictors(), -has_role("ID")) %>%
+    step_zv(all_predictors()) %>%
     # Replace novel levels with "new"
     step_novel(all_of(cat_vars), -has_role("ID")) %>%
     # Replace NA in factors with "unknown"
     step_unknown(all_of(cat_vars), -has_role("ID")) %>%
-    # Perform categorical embedding
-    embed::step_lencode_glm(
-      all_of(cat_vars), -has_role("ID"),
-      outcome = vars(meta_sale_price)
-    ) %>%
+    # Dummify categorical predictors
+    step_dummy(all_nominal_predictors(), -has_role("ID"), one_hot = TRUE) %>%
```

Comment:
Instead of using `embed` (expensive!) we just one-hot encode the remaining location categoricals.